storey/flow.py

```diff
@@ -643,17 +647,23 @@ async def _cleanup(self):
     async def _lazy_init(self):
         pass
 
-    async def _safe_process_event(self, event):
-        if event._awaitable_result:
+    async def _process_event_with_retries(self, event):
+        times_attempted = 0
+        max_attempts = self._retries + 1
+        while True:
             try:
                 return await self._process_event(event)
-            except BaseException as ex:
-                none_or_coroutine = event._awaitable_result._set_error(ex)
```

Comment:
Why have you removed it?

Reply:
So the answer is a bit nonobvious. If this throws an exception, it will be caught inside of the `_worker` method, where `_set_error` is already called for any `BaseException`. So this line was redundant and potentially could even cause termination issues (depending on the evolving semantics of completions).